uaclient/entitlements/tests/test_livepatch.py

```diff
@@ -0,0 +1,220 @@
+        cfg.write_cache('machine-access-livepatch',
+                        LIVEPATCH_RESOURCE_ENTITLED)
+        with mock.patch('sys.stdout', new_callable=StringIO) as m_stdout:
+            entitlement = LivepatchEntitlement(cfg)
+            self.assertFalse(entitlement.can_enable())
+        msg = 'Cannot install Livepatch on a container\n'
+        assert msg == m_stdout.getvalue()
+
+
+class TestLivepatchEntitlementEnable(TestCase):
+
+    with_logs = True
+
+    mocks_install = [mock.call(
+        ['snap', 'install', 'canonical-livepatch'], capture=True)]
+    mocks_config = [
+        mock.call(
+            ['/snap/bin/canonical-livepatch', 'config',
+             'remote-server=https://alt.livepatch.com'], capture=True),
+        mock.call(
+            ['/snap/bin/canonical-livepatch', 'enable', 'TOKEN'], capture=True)
+    ]
+
+    @mock.patch(M_PATH + 'LivepatchEntitlement.can_enable', return_value=False)
+    def test_enable_false_when_can_enable_false(self, m_can_enable):
+        """When can_enable returns False enable returns False."""
+        tmp_dir = self.tmp_dir()
+        cfg = config.UAConfig(cfg={'data_dir': tmp_dir})
+        cfg.write_cache('machine-token', LIVEPATCH_MACHINE_TOKEN)
+        cfg.write_cache('machine-access-livepatch',
+                        LIVEPATCH_RESOURCE_ENTITLED)
+        entitlement = LivepatchEntitlement(cfg)
+        with mock.patch('sys.stdout', new_callable=StringIO) as m_stdout:
+            self.assertFalse(entitlement.enable())
+        assert '' == self.logs  # No additional logs on can_enable == False
+        assert [mock.call()] == m_can_enable.call_args_list
+
+    @mock.patch('uaclient.util.subp')
+    @mock.patch('uaclient.util.which', return_value=False)
+    @mock.patch(M_PATH + 'LivepatchEntitlement.can_enable', return_value=True)
+    def test_enable_installs_livepatch_snap_when_absent(
+            self, m_can_enable, m_which, m_subp):
+        """Install canonical-livepatch snap when not present on the system."""
+        tmp_dir = self.tmp_dir()
+        cfg = config.UAConfig(cfg={'data_dir': tmp_dir})
+        cfg.write_cache('machine-token', LIVEPATCH_MACHINE_TOKEN)
+        cfg.write_cache('machine-access-livepatch',
+                        LIVEPATCH_RESOURCE_ENTITLED)
+        entitlement = LivepatchEntitlement(cfg)
+
+        with mock.patch('sys.stdout', new_callable=StringIO) as m_stdout:
+            self.assertTrue(entitlement.enable())
+        assert self.mocks_install + self.mocks_config in m_subp.call_args_list
+        msg = ('Installing canonical-livepatch snap...\n'
+               'Canonical livepatch enabled.\n')
+        assert msg == m_stdout.getvalue()
+
+    @mock.patch('uaclient.util.subp')
+    @mock.patch('uaclient.util.which', return_value='/found/livepatch')
+    @mock.patch(M_PATH + 'LivepatchEntitlement.can_enable', return_value=True)
+    def test_enable_installs_and_configures_livepatch_snap_when_present(
```

Comment:
This test name is a little off, we're testing that it _doesn't_ install it if it's already present.

Reply:
fixed name and docstr